Exercise_1.py

```diff
@@ -1,12 +1,24 @@
+"""
+Runtime Complexity:
+1)O(n) as we scan through all the elements in the list to check whether the target is present in the list.
+Space Complexity:
+1) O(n) - as the loop would go 'n' iterative calls
+
+- Yes, the code worked on leetcode.
+- Idea - A loop to check if the target element is in the list, and if it's present then we return the index of it.
+"""
 # Python code to implement iterative Binary  
 # Search. 
   
 # It returns location of x in given array arr  
 # if present, else returns -1 
-def binarySearch(arr, l, r, x): 
+def binarySearch(arr, l, r, x):            #array, start, end, target element are passed as parameters.
   
   #write your code here
-  
+  for i in range(l,r):                    #loop from start to end to check if the target is present in the list
```

Comment:
It is linear search . You needed to implement binary search. Each time you need to calculate mid value and accordingly move your left and right pointers.